Scripts/RemediationScripts/Remediate-EnableAppServicEendToEndEncryption.ps1

```diff
@@ -0,0 +1,943 @@
+    }
+    # Setting up context for the current Subscription.
+    $context = Set-AzContext -SubscriptionId $SubscriptionId -ErrorAction Stop
+    if (-not($AutoRemediation)) {
+        Write-Host "Subscription Name: [$($context.Subscription.Name)]"
+        Write-Host "Subscription ID: [$($context.Subscription.SubscriptionId)]"
+        Write-Host "Account Name: [$($context.Account.Id)]"
+        Write-Host "Account Type: [$($context.Account.Type)]"
+        Write-Host $([Constants]::SingleDashLine)
+    }
+    Write-Host "To enable End-to-end TLS encryption for App Services in a Subscription, Contributor or higher privileges on the App Services are required." -ForegroundColor $([Constants]::MessageType.Warning)
+    Write-Host $([Constants]::SingleDashLine)
+    Write-Host "[Step 2 of 4] Fetch all App Services"
+    Write-Host $([Constants]::SingleDashLine)
+
+     # list to store resource details.
+    $AppServicesDetails = @()
+   
+
+    # To keep track of remediated and skipped resources
+    $logRemediatedResources = @()
+    $logSkippedResources = @()
+    $appServicesAPI = "https://management.azure.com/subscriptions/$subscriptionId/providers/Microsoft.Web/sites?api-version=2024-04-01" 
+
+    #Control id for the control
+    $controlIds = "Azure_AppService_DP_Configure_EndToEnd_TLS"
+
+    if($AutoRemediation)
+    {
+        if(-not (Test-Path -Path $Path))
+        {
+            Write-Host "File containing failing controls details [$($Path)] not found. Skipping remediation..." -ForegroundColor $([Constants]::MessageType.Error)
+            Write-Host $([Constants]::SingleDashLine)
+            return
+        }
+        Write-Host "Fetching all App Services failing for the [$($controlIds)] control from [$($Path)]..." -ForegroundColor $([Constants]::MessageType.Info)
+        Write-Host $([Constants]::SingleDashLine)
+        $controlForRemediation = Get-content -path $Path | ConvertFrom-Json
+        $subId = $controlForRemediation.SubscriptionId
+        $controls = $controlForRemediation.ControlRemediationList
+        $resourceDetails = $controls | Where-Object { $controlIds -eq $_.ControlId };
+        $validResources = $resourceDetails.FailedResourceList | Where-Object {![String]::IsNullOrWhiteSpace($_.ResourceId)}
+        if(($resourceDetails | Measure-Object).Count -eq 0 -or ($validResources | Measure-Object).Count -eq 0)
+        {
+            Write-Host "No App Services found in input json file for remediation." -ForegroundColor $([Constants]::MessageType.Error)
+            Write-Host $([Constants]::SingleDashLine)
+            return
+        }
+        $validResources | ForEach-Object { 
+            try
+            {
+                $appServiceResourceAPI = "https://management.azure.com/subscriptions/$subId/resourceGroups/$($_.ResourceGroupName)/providers/Microsoft.Web/sites/$($_.Name)?api-version=2024-04-01" 
+                $AppServiceDetail = Fetch-API -Method Get -Uri $appServiceResourceAPI -ErrorAction Stop
```

Comment:
SHould not we filter for applicable kinds?

Reply:
Done